[v2] Code Sessions docs (#36064)
DataDog/documentation: content/en/bits_ai/bits_ai_dev_agent/_index.md

Bits AI Dev Agent is a generative AI coding assistant that uses observability data from Datadog to automatically diagnose and fix issues in your code. It integrates with GitHub to create production-ready pull requests, iterates on fixes using CI logs and developer feedback, and draws on multiple Datadog products to generate contextual fixes.

> Each time the Dev Agent investigates an issue or generates a fix, it creates a [**code session**][18], which captures the agent's analysis, actions, and any resulting code changes across supported Datadog products.

Author: ```suggestion
Each time the Dev Agent investigates an issue or generates a fix, it creates a [**code session**][19], which captures the agent's analysis, actions, and any resulting code changes across supported Datadog products.
```

To get started with Bits AI Dev Agent, [set up the GitHub integration][6] and complete any additional configuration. Then, [start your first code session][18].

## Code sessions
A code session captures a segment of work with the Bits AI Dev Agent, including its analysis and code changes. Start, view, and manage your sessions at **Bits AI** > **Dev Agent** > [**Code Sessions**][7].

{{< img src="bits_ai/dev_agent/code_sessions_overview.png" alt="A tab titled 'Code Sessions' shows a text field with suggestions underneath" style="width:100%;" >}}

### Start a code session

> After [completing setup][6], to start a code session:

Reviewer: Now that I see this again, we might want to make it clearer that these aren't sequential steps (they're not numbered, but that might still be too subtle for some readers):
```suggestion
After [completing setup][6], do one of the following to start a code session:
```

Author: good call, thanks! implemented

- Enter a freeform prompt at [**Code Sessions**][7]: enter a custom prompt or generate one by clicking a **Suggestions** or **Proactive Fixes** card
- Invoke Bits AI Dev Agent in a [supported Datadog product][9]

A code session can also be created when another Bits AI agent (like [Bits Assistant][16] or [Bits AI SRE][17]) hands off a coding task to the Dev Agent.

### View and manage code sessions

> On [Code Sessions][7], view your past sessions in the **My Sessions** panel. A session appears here if you initiated it or interacted with it in some way, like participating in the conversation or creating an associated PR.

Author: ```suggestion
On **[Code Sessions][7]**, view your past sessions in the **My Sessions** panel. A session appears here if you initiated it or interacted with it in some way, like participating in the conversation or creating an associated PR.
```

> Click a session to view its details and continue working with the Dev Agent. To remove a session from your **My Sessions** list, click {{< img src="bits_ai/dev_agent/archive_icon.png" inline="true" style="width:24px">}} (**Archive for everyone**) or {{< img src="bits_ai/dev_agent/unwatch_icon.png" inline="true" style="width:24px">}} (**Unwatch session**).

Reviewer: You could also use the [icon tag](https://datadoghq.atlassian.net/wiki/spaces/docs4docs/pages/1182958054/Markdown+Shortcode+Reference#Icons) (`<i class="icon-name"></i>`) for inline images. I think one advantage is the source images come directly from the UI design library, so if the UI gets changed, the icons in the docs are automatically updated too (in theory).

Author: whoa, thank you so much for this tip! I used the icons and they look great in the preview

Bits AI Dev Agent is a generative AI coding assistant that uses observability data from Datadog to automatically diagnose and fix issues in your code. It integrates with GitHub to create production-ready pull requests, iterates on fixes using CI logs and developer feedback, and draws on multiple Datadog products to generate contextual fixes.

Each time the Dev Agent investigates an issue or generates a fix, it creates a [**code session**][18], which captures the agent's analysis, actions, and any resulting code changes across supported Datadog products.

> To get started with Bits AI Dev Agent, follow the instructions on [Setup][6]. Then, [create your first code session][18].

Reviewer: The first link could be a little more descriptive of what's involved. Most of the _required_ setup seems to be installing the GitHub integration so we could highlight that and mention other config too. Also, it seems like the user "starts" a code session (vs. something they would more actively create, like a new user account, etc.)

```suggestion
To get started with Bits AI Dev Agent, [set up the GitHub integration][6] and complete any additional configuration. Then, [start your first code session][18].
```

Author: good calls! I went back and forth between "Start" and "Create" and I think you're right about the former better describing the action. gonna change the relevant section header below, too!

## Code sessions

> A code session captures a segment of work with the Bits AI Dev Agent, including its analysis and code changes. Create, view, and manage your sessions on **Bits AI** > **Dev Agent** > [**Code Sessions**][7].

Reviewer: I think it's more idiomatic in our docs to use "at" for a location in the UI, but it can vary case by case:
```suggestion
A code session captures a segment of work with the Bits AI Dev Agent, including its analysis and code changes. Create, view, and manage your sessions at **Bits AI** > **Dev Agent** > [**Code Sessions**][7].
```

{{< img src="bits_ai/dev_agent/code_sessions_overview.png" alt="A tab titled 'Code Sessions' shows a text field with suggestions underneath" style="width:100%;" >}}

### Create a code session
After [completing setup][6], you can start a code session:
- By entering a freeform prompt on [**Code Sessions**][7]
  - Enter a custom prompt or generate one by clicking a **Suggestions** card. (If the Dev Agent has created proactive fixes for any of your repositories, a **Proactive Fixes** section appears instead.)

> - By invoking Bits AI Dev Agent in a [supported Datadog product][9]

Reviewer: No need for a sub-bullet if there's just one item, and the background explanation feels unnecessary if we're just trying to explain how to start a code session. Slight edit to be more action-oriented:
```suggestion
After [completing setup][6], to start a code session:
- Enter a freeform prompt at [**Code Sessions**][7]: Enter a custom prompt or generate one by clicking a **Suggestions** or **Proactive Fixes** card
- Invoke Bits AI Dev Agent in a [supported Datadog product][9]
```

Author: ah, nice! I think you're right about the Proactive Fixes context being superfluous. one nit: I think we should lowercase the "enter" after the colon [based on this guidance](https://datadoghq.atlassian.net/wiki/spaces/WRITING/pages/5341577434/Capitalization#Capitalization-after-colons)

[6]: /bits_ai/bits_ai_dev_agent/setup/
[7]: https://app.datadoghq.com/code
[8]: /bits_ai/bits_ai_sre/
[9]: #supported-datadog-products
[10]: /profiler/automated_analysis/
[12]: /containers/
[13]: /containers/bits_ai_kubernetes_remediation
[14]: https://app.datadoghq.com/code/settings
[15]: /security/code_security/static_analysis/ai_enhanced_sast/
[16]: /bits_ai/bits_assistant/
[17]: /bits_ai/bits_ai_sre/

> [18]: #start-a-code-session

Author: ```suggestion
[18]: #start-a-code-session
[19]: #code-sessions
```
